guides/source/active_job_basics.md

```diff
@@ -131,77 +119,330 @@ That's it!
+end
 ```
 
-[`config.active_job.queue_adapter`]: configuring.html#config-active-job-queue-adapter
+This ensures that only one job for a given contact can run at a time, regardless of the job class.
 
-### Starting the Backend
+Read more about [Concurrency Controls in the Solid Queue documentation](https://github.com/rails/solid_queue?tab=readme-ov-file#concurrency-controls).
 
-Since jobs run in parallel to your Rails application, most queuing libraries
-require that you start a library-specific queuing service (in addition to
-starting your Rails app) for the job processing to work. Refer to library
-documentation for instructions on starting your queue backend.
+### Failed Jobs and Retries
 
-Here is a noncomprehensive list of documentation:
+Solid Queue doesn’t provide automatic retries; it relies on Active Job for that. Failed jobs will be stored in the `solid_queue_failed_executions` table. You can inspect and retry them manually, for example:
 
-- [Sidekiq](https://github.com/mperham/sidekiq/wiki/Active-Job)
-- [Resque](https://github.com/resque/resque/wiki/ActiveJob)
-- [Sneakers](https://github.com/jondot/sneakers/wiki/How-To:-Rails-Background-Jobs-with-ActiveJob)
-- [Queue Classic](https://github.com/QueueClassic/queue_classic#active-job)
-- [Delayed Job](https://github.com/collectiveidea/delayed_job#active-job)
-- [Que](https://github.com/que-rb/que#additional-rails-specific-setup)
-- [Good Job](https://github.com/bensheldon/good_job#readme)
-- [Solid Queue](https://github.com/rails/solid_queue?tab=readme-ov-file#solid-queue)
+```ruby
+failed_execution = SolidQueue::FailedExecution.find(1)
+failed_execution.error # inspect the error
+
+failed_execution.retry  # Re-enqueues the job
+failed_execution.discard  # Deletes the failed job
+```
+
+For better tracking, you can use a tool like [`mission_control-jobs`](https://github.com/rails/mission_control-jobs) to manage failed jobs. Read more about [Failed Jobs and Retries in the Solid Queue Documentation](https://github.com/rails/solid_queue?tab=readme-ov-file#failed-jobs-and-retries).
+
+### Error Reporting on Jobs
+
+If your error tracking service doesn’t automatically report job errors, you can manually hook into Active Job to report exceptions. For example, you can add a `rescue_from` block in `ApplicationJob`:
+
+```ruby
+class ApplicationJob < ActiveJob::Base
+  rescue_from(Exception) do |exception|
+    Rails.error.report(exception)
+    raise exception
+  end
+end
+```
+
+If you use ActionMailer, you’ll need to handle errors for `MailDeliveryJob` separately:
+
+```ruby
+class ApplicationMailer < ActionMailer::Base
+  ActionMailer::MailDeliveryJob.rescue_from(Exception) do |exception|
+    Rails.error.report(exception)
+    raise exception
+  end
+end
+```
+
+Read more [Error Reporting on Jobs in the Solid Queue Documentation](https://github.com/rails/solid_queue?tab=readme-ov-file#error-reporting-on-jobs).
+
```

Comment:
This section already describes how Active Job can be used to handle errors. No need to link to Solid Queue, I think.
```suggestion
```